tests/utils/test_downstream.py

```diff
@@ -0,0 +1,134 @@
+        "minimal_mastering_unified_dataset.cluster_feedback",
+        "minimal_mastering_unified_dataset.internal_comments",
+        "minimal_mastering_unified_dataset.global_record_id",
+    }
+    assert DEPENDENT_DATASETS_BY_NAME.issubset(set([d.name for d in downstream_datasets]))
+
+
+@mock_api(enforce_online_test=ONLINE_TEST)
+def test_get_downstream_projects():
+    client = tamr_toolbox.utils.client.create(**CONFIG["toolbox_test_instance"])
+    dataset = client.datasets.by_resource_id(MASTERING_UNIFIED_DATASET_ID)
+    downstream_projects = tamr_toolbox.utils.downstream.projects(
+        dataset, include_dependencies_by_name=True
+    )
+    assert len(downstream_projects) == 2
+    assert (
+        str(downstream_projects) == "[tamr_unify_client.project.resource.Project(relative_id="
+        "'projects/1', name='minimal_mastering', "
+        "type='DEDUP'), tamr_unify_client.project.resource.Project"
+        "(relative_id='projects/2', name='minimal_golden_records', "
+        "type='GOLDEN_RECORDS')]"
+    )
+
+
+@mock_api(enforce_online_test=ONLINE_TEST)
+def test_get_downstream_datasets_for_source_dataset():
+    client = tamr_toolbox.utils.client.create(**CONFIG["toolbox_test_instance"])
+    source_dataset = client.datasets.by_resource_id(SOURCE_DATASET_ID)
+    unified_dataset = client.datasets.by_resource_id(MASTERING_UNIFIED_DATASET_ID)
+    source_downstream_datasets = tamr_toolbox.utils.downstream.datasets(source_dataset)
+    unified_downstream_datasets = tamr_toolbox.utils.downstream.datasets(unified_dataset)
+    source_downstream_datasets_names = set([d.name for d in source_downstream_datasets])
+    unified_downstream_datasets_names = set([d.name for d in unified_downstream_datasets])
+    # All downstream datasets of unified dataset must be downstream datasets of the source dataset
+    assert unified_downstream_datasets_names.issubset(source_downstream_datasets_names)
+
+
+@mock_api(enforce_online_test=ONLINE_TEST)
+def test_get_downstream_datasets_for_source_dataset_suggest_name():
+    client = tamr_toolbox.utils.client.create(**CONFIG["toolbox_test_instance"])
+    source_dataset = client.datasets.by_resource_id(SOURCE_DATASET_ID)
+    unified_dataset = client.datasets.by_resource_id(MASTERING_UNIFIED_DATASET_ID)
+    source_downstream_datasets = tamr_toolbox.utils.downstream.datasets(
+        source_dataset, include_dependencies_by_name=True
+    )
+    unified_downstream_datasets = tamr_toolbox.utils.downstream.datasets(
+        unified_dataset, include_dependencies_by_name=True
+    )
+    source_downstream_datasets_names = set([d.name for d in source_downstream_datasets])
+    unified_downstream_datasets_names = set([d.name for d in unified_downstream_datasets])
+    # All downstream datasets of unified dataset must be downstream datasets of the source dataset
+    assert unified_downstream_datasets_names.issubset(source_downstream_datasets_names)
```

Comment:
Same question as before, why are we only checking for a subset?

Reply:
list of downstream datasets of unified dataset should be a subset of downstream datasets of a source dataset, not equal

Reply:
Are there any additional things we can check for in addition to this subset? Checking for a subset is a bit loose. If the results were [a,b,c,d,e,f,g] and [e] that would pass issubset even if the expected results were more like [a,b,c,d,e,f,g] and [a,b,c]

Reply:
Testing the list of downstream datasets for the unified dataset is already done in the previous test. The previous test already confirms that `unified_downstream_dataset_names == [a,b,c]`. This test aims at testing the downstream datasets for a source dataset, particularly the logic `if datasetA is downstream of dataset B then downstream_datasets(datasetA) is a subset of downstream_datasets(datasetB)`. 

I can also test equal for the list of downstream datasets for the source dataset. But if any new project uses the same source dataset as input dataset, the test set will need to be modified. Since this won't be a problem unless online test is run, a strict comparison is not a big deal either. I will make that change.

Reply:
@nguyentr17 Your logic makes sense. You can leave ti as subset. Thanks for explaining

Reply:
great, I reverted to the test for subset to make it more robust. Thanks!